xdsl/tools/xdsl-opt

```diff
@@ -15,6 +17,43 @@ class OptMain(xDSLOptMain):
     def register_all_arguments(self, arg_parser: argparse.ArgumentParser):
         super().register_all_arguments(arg_parser)
 
+        #arg_parser.add_argument(
+        #    "--check-coverage",
+        #    default=False,
+        #    action='store_true',
+        #    help="Generate the xDSL code coverage for this run.")
+
+    def run(self):
+        # In order to find the `.coverage` file, as well as the `.coveragerc`,
+        # we need to step up to the `xdsl` directory. Notice that lit runs tests
+        # within the folder their are contained in.
+        while (os.getcwd().split('/')[-1] != 'xdsl'):
+            os.chdir('../')
+
+        data_file = ".coverage"
+        cov = coverage.Coverage(check_preimported=True,
+                                data_file=data_file,
+                                data_suffix='',
+                                config_file='.coveragerc',
+                                source_pkgs=['xdsl'],
```

Comment:
This argument should make it s.t. the coverage is only reported for files within the `xdsl/` directory. To me, this makes sense as it is the thing we want to test. However, it is different to what the configuration for pytest currently does (can be found in `.coveragerc`). Is there a reason for that? Maybe @georgebisbas

I will look into synchronizing the two s.t. these arguments can be loaded form the `.coveragerc` directly.

Reply:
The fact that the covered files is reduced in this PR might be the reason why the test coverage drops in the PR.

Reply:
Or it might be the case that through specifying a source, coverage can now understand that some of the files are not actually being executed.

Reply:
Yes, it seems that this PR covers 37 files, whereas the old coverage data only covered 29 files.

Reply:
I have synchronized the two ways of generating coverage data. Now, only the `xdsl` folder is considered, which seems to hurt our coverage further.